Scripts/script-CommonServerPython.yml

```diff
@@ -28,31 +28,31 @@ script: |-
   thresholds = {'xfeScore': 4, 'vtPositives': 10, 'vtPositiveUrlsForIP': 30}
   dbotscores = {'Critical': 4, 'High': 3, 'Medium': 2,'Low': 1, 'Unknown': 0, 'Informational': 0.5}
 
-  SAVE_PROXIES = {}
-
-  def disable_proxy():
+  def handle_proxy(proxy_param_name='proxy', checkbox_default_value=False):
       """
-          Disable routing traffic through the system proxy.
+          Handle logic for routing traffic through the system proxy.
           Should usually be called at the beginning of the integration, depeding on proxy checkbox state.
-          :rtype: ``None``
-          :return: None
-      """
-      global SAVE_PROXIES
-      for k in ('HTTP_PROXY', 'HTTPS_PROXY', 'http_proxy', 'https_proxy'):
-          SAVE_PROXIES[k] = os.environ[k]
-          del os.environ[k]
 
-  def enable_proxy():
-      """
-          Re-enable routing traffic through the system proxy (for container re-use).
-          Should usually be called in a "finally" block at the bottom of the integration.
-          :rtype: ``None``
-          :return: None
-      """
-      global SAVE_PROXIES
-      for k, v in SAVE_PROXIES.iteritems():
-          os.environ[k] = v
-      SAVE_PROXIES = {}
+          :type proxy_param_name: ``string``
+          :param proxy_param_name: name of the "use system proxy" integration parameter
+
+          :type checkbox_default_value: ``bool``
+          :param checkbox_default_value: Default value of the proxy param checkbox
+
+          :rtype: ``dict``
+          :return: proxies dict for the 'proxies' parameter of 'requests' functions
+      """
+      proxies = {}
+      if demisto.params().get(proxy_param_name, checkbox_default_value):
+          proxies = {
+              'http': os.environ.get('HTTP_PROXY') or os.environ.get('http_proxy', ''),
+              'https': os.environ.get('HTTPS_PROXY') or os.environ.get('https_proxy', '')
+          }
+      else:
+          for k in ('HTTP_PROXY', 'HTTPS_PROXY', 'http_proxy', 'https_proxy'):
+              if k in os.environ:
+                  del os.environ[k]
```

Comment:
when will this be restored?

Reply:
@yaakovi the restore is done in the python loop. No need to perform this logic in common server.